keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,920 @@
+Resource Allocation (DRA) KEP [3], offers the ability to transition compute 
+resource allocation behavior from being a cluster-admin dominated configuration
+to one that allows users with precise compute requirements to articulate the
+compute attributes that they need. 
+
+In several domains such Streaming, Telco, HPC, and AI/ML, users require
+fine-grained compute resource control to maximize their key performance 
+indicators.  They need an API option that can be made available through
+upstream Kubernetes components and specific compute resource allocation 
+plugins.
+
+##### Example 1: Attributed-based resource specification
+
+One realization of such an API can be achieved by an attribute-based 
+mechanism for compute resources. The attribute-based compute resource 
+request model is a step towards improved semantic workload portability 
+that can be tried in a progressive manner with CCI. 
+
+In the following, there is an example of a novel approach that could be 
+fulfilled with the CCI architecture. This example presents a set of
+per-core attributes in a config-map form that enable precision beyond a policy-only based approach: 
+
+    # A resource request which consists of:
+    # 2 exclusive cores (no other processes running on them)
+    # 	* the cores shall be siblings
+    # 	* they shall run at 2.1 ghz
+    # 	* application assigned has the highest priority on the core
+    # 	* there are no IRQs on the core
+    # 6 exclusive cores(no other processes running on them)
+    # 4 shared cores(other processes can run on them) 
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: example-compute-claim-parameters
+      namespace: default
+    data:
+      cores: |
+    # number of core type 1
+    2
+    # number of core type 2
+    6
+    # number of core type 3
+    4
+      compute_attributes: |
+        # Attributes of core type 1
+        exclusive, smt-sibling-required, 2.1Ghz, 1.0 priority, no-irq
+        # Attributes of core type 2
+        exclusive
+        # Attributes of core type 3
+        shared
+
+##### Example 2.: Policy-based resource specification
+
+Another option how a resource allocation configuration for Kubernetes
+Pods could be achieved is through a policy mechanism. In this case the
+policy will be applied to all Pods to be processed. The configuration
+of the policy can happen directly in a CCI Compute Resource Driver Plugin
+as input argument in the yaml deployment. This methodology can be used to
+realize methodologies similar to the static CPU manager policy or other policies.
+
+##### Resource Manager Architecture
+
+The extension consists of an optional CCI driver inside Pod spec which
+indicates if a CCI driver exists, then it shall process compute resources.
+The association betweed pod and CCI Driver can be also realised as a resource class 
+argument (will be considered as an option during the implementation). 
```

Comment:
I would really like us to nail down the design before going into the implementation stage. Leaving things open ended is not ideal. Let's clearly articulate the method of indicating that a CCIDriver is going to be responsible for allocating resources. I understand that we discussed that if the feature gate is enabled, CCIDriver takes over resource allocation but saying that we would either have an optional field in the pod spec or specify via a resource class is too open ended IMO. In previous discussion with @klueska reference the use of `resourceClaims` came up as well? Are we using ResourceClasses and then the resource request specified as resource claims or a separate CCIDriver field in the podspec? How are the parameters or attributes specified in the latter?

Reply:
as discussed previously in alpha stage the CCI Driver association is not needed, we will control that via the feature gate . In post-alpha resource class argument is one option how to deal with that . If i undertood Kevin correctly we make this api decision in post-alpha stage.